src/miniscript/decode.rs

```diff
@@ -195,6 +196,9 @@ impl<Pk: MiniscriptKey, Ctx: ScriptContext> TerminalStack<Pk, Ctx> {
         let left = self.pop().unwrap();
         let right = self.pop().unwrap();
 
+        Ctx::check_ms_validity(&left)?;
+        Ctx::check_ms_validity(&right)?;
+
```

Comment:
We should have checks on the wrapped_ms instead of children. It is possible that the children did not reach stack limit, but the combination did. 

Similar to how it is done for check_frag_validity.

Reply:
Actually, we can't. We need to call it on a Miniscript not a Terminal.

Reply:
Terminal and Miniscript are equivalent. Miniscript is Arc to Terminal. It is cheap to create and clone Arcs

Reply:
The return values of all functions `reduce0/reduce1/reduce2` are all Miniscripts corresponding to the terminals.
We can change the existing check_frag_validity to take in &Miniscript as an argument and use the `node` field to get the terminals. I think this will simplify the code.

Reply:
Right. This should be part of #149 anyways.